diff --git a/Telegram/SourceFiles/data/components/sponsored_messages.cpp b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
--- a/Telegram/SourceFiles/data/components/sponsored_messages.cpp	(revision c795ea461d1e7c3b71fb5f7cd185fdfede331b9a)
+++ b/Telegram/SourceFiles/data/components/sponsored_messages.cpp	(date 1764301666650)
@@ -248,8 +248,7 @@
 }
 
 bool SponsoredMessages::canHaveFor(not_null<HistoryItem*> item) const {
-	return item->history()->peer->isBroadcast()
-		&& item->isRegular();
+	return false；
 }
 
 bool SponsoredMessages::isTopBarFor(not_null<History*> history) const {
